extension/src/openvic-extension/singletons/PopulationMenu.cpp

```diff
@@ -525,6 +530,68 @@ void MenuSingleton::_population_menu_sort_pops() {
 	}
 
 	emit_signal(_signal_population_menu_pops_changed());
+
+	return true;
+}
+
+Error MenuSingleton::population_menu_update_locale_sort_cache() {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, FAILED);
+	InstanceManager const* instance_manager = game_singleton->get_instance_manager();
+	ERR_FAIL_NULL_V(instance_manager, FAILED);
+
+	std::vector<String> localised_items;
+	std::vector<size_t> sorted_items;
+
+	const auto generate_sort_cache = [this, &localised_items, &sorted_items]<HasGetIdentifier T>(
+		IndexedMap<T, size_t>& cache, std::vector<T> const& items
+	) {
+		localised_items.resize(items.size());
+		sorted_items.resize(items.size());
+
+		for (size_t idx = 0; idx < items.size(); ++idx) {
+			String identifier = std_view_to_godot_string(items[idx].get_identifier());
+			if constexpr (std::is_same_v<T, ProvinceInstance>) {
+				identifier = GUINode::format_province_name(identifier);
```

Comment:
For types that are displayed in the UI (countries, provinces, pop types, etc) maybe we should add a `get_display_name()` method that you can use here instead of type checking.

Reply:
This is a constexpr check, its conditionally compiled, it does not exist if T is not ProvinceInstance.

Reply:
As Spartan said, this is a compile time check so it has no effect at runtime for types other than `ProvinceInstance`. Although we could solve this kind of scenario by giving each type a function, in this specific case it would be more trouble than it's worth as only `ProvinceInstance` has a different localisation key from its identifier (split states add a country adjective to their localised name, but that's a separate key which has to be handled at the GDExtension level (in this repo) and so couldn't be solved by adding functions to these SIM repo structs).